swift/IceStorm/clock/Sources/Publisher/main.swift

```diff
@@ -18,114 +18,106 @@ enum Option: String {
+let ctrlCHandler = Ice.CtrlCHandler()
 
-        let communicator = try Ice.initialize(args: &args, configFile: "config.pub")
-        defer {
-            communicator.destroy()
-        }
+// Create an Ice communicator to initialize the Ice runtime.
+var args = [String](CommandLine.arguments.dropFirst())
+let communicator = try Ice.initialize(args: &args, configFile: "config.pub")
+defer {
+    communicator.destroy()
+}
 
-        var option: Option = .none
-        var topicName = "time"
-
-        for var i in 0 ..< args.count {
-            let oldoption = option
-
-            if args[i].starts(with: "--") {
-                guard let o = Option(rawValue: args[i]) else {
-                    usage()
-                    return 1
-                }
-                option = o
-            } else {
-                topicName = args[i]
-                i += 1
-                break
-            }
-
-            if oldoption != option, oldoption != .none {
-                usage()
-                return 1
-            }
-        }
+// Parse command-line options.
 
-        guard let base = try communicator.propertyToProxy("TopicManager.Proxy"),
-            let manager = try await checkedCast(prx: base, type: IceStorm.TopicManagerPrx.self) else {
-            print("invalid proxy")
-            return 1
-        }
+var option: Option = .none
+var topicName = "time"
 
-        //
-        // Retrieve the topic.
-        //
-        var topic: IceStorm.TopicPrx!
-        do {
-            topic = try await manager.retrieve(topicName)
-        } catch is IceStorm.NoSuchTopic {
-            do {
-                topic = try await manager.create(topicName)
-            } catch is IceStorm.TopicExists {
-                print("temporary error. try again.")
-                return 1
-            }
-        }
+for var i in 0..<args.count {
+    let oldOption = option
 
-        //
-        // Get the topic's publisher object, and create a Clock proxy with
-        // the mode specified as an argument of this application.
-        //
-        guard var publisher = try await topic.getPublisher() else {
-            print("Error getting publisher proxy")
-            return 1
+    if args[i].starts(with: "--") {
+        guard let o = Option(rawValue: args[i]) else {
+            usage()
+            exit(1)
         }
+        option = o
+    } else {
+        topicName = args[i]
+        i += 1
+        break
+    }
 
-        switch option {
-        case .datagram:
-            publisher = publisher.ice_datagram()
-        case .twoway:
-            // Do Nothing
-            break
-        default:
-            publisher = publisher.ice_oneway()
-        }
+    if oldOption != option, oldOption != .none {
+        usage()
+        exit(1)
+    }
+}
 
-        let clock = uncheckedCast(prx: publisher, type: ClockPrx.self)
+// Create the topic manager proxy.
+let manager = try uncheckedCast(
```

Comment:
Isn't there a generic propertyToProxy for Swift?

Reply:
There is currently only:
```
Ice/Communicator.swift
61:    func propertyToProxy(_ property: String) throws -> ObjectPrx?
```